sap_business_document_processing/document_information_extraction_client/dox_api_client.py

```diff
@@ -633,3 +640,247 @@ def post_confirm_document(self, document_id, data_for_retraining=False) -> dict:
+        """
+        url = SCHEMAS_VERSIONS_ENDPOINT.format(schemaId=schema_id) + "?clientId={}".format(client_id)
+        return self.get(url)
+
+    def get_schema_version_details(self, schema_id, version, client_id):
+        """
+        GET /schemas/{schemaId}/versions/{version}
+        """
+        response = self._get_schema_version_details(schema_id, version, client_id)
+        return response
+
+    def _get_schema_version_details(self, schema_id, version, client_id):
+        """
+        Retrieve Version Details of Schema for a Client
+        """
+        url = SCHEMAS_VERSION_UUID_ENDPOINT.format(schemaId=schema_id, versionId=version) + "?clientId={}" \
+            .format(client_id)
+        return self.get(url)
+
+    def create_schema_with_fields(self, client_id, payload, model_type, item_fields=None):
+        """
+        Create schema along with header fields and line items
+        Sample payload for posting fields:
+            fields_payload = [
+                {
+                    "name": "documentNumber",
+                    "description": "",
+                    "label": "",
+                    "datatype": "",
+                    "is_line_item": False
+                },
+                {
+                    "name": "netAmount",
+                    "description": "",
+                    "label": "",
+                    "datatype": "",
+                    "is_line_item": True
+                }
+            ]
+            model_type = "defaultWorkflow"
+            setupTypeVersion = "1.0.0"
+            response = create_schema_with_fields(external_client, schema_payload, model_type + setupTypeVersion,
+                                                fields_payload)
+        """
+        if model_type not in SUPPORTED_MODEL_TYPES:
+            error_msg = 'Invalid model. Valid models are {}.'.format(str(SUPPORTED_MODEL_TYPES))
+            raise DoxApiInvalidDataProvidedError(error_msg)
+        file_path = self._get_default_extractor_fields(model_type)
+        schema_fields_payload = create_payload_for_schema_fields(model_type, item_fields)
+        if model_type in [MODEL_TYPE_DEFAULT + SETUP_TYPE_VERSION_1, MODEL_TYPE_DEFAULT + SETUP_TYPE_VERSION_2]:
+            os.remove(file_path)
+        response = self.post_schema_configuration(payload)
+        schema_id = response.json()["id"]
+        return self.post_schema_version_fields(schema_fields_payload, schema_id, '1', client_id)
+
+    def _get_default_extractor_fields(self, model_type):
+        if model_type in [MODEL_TYPE_DEFAULT + SETUP_TYPE_VERSION_1, MODEL_TYPE_DEFAULT + SETUP_TYPE_VERSION_2]:
+            capabilities = self.get_capabilities()
+            file_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), DEFAULT_EXTRACTOR_FIELDS_FILE_PATH)
+            try:
+                with open(file_path, 'w') as file:
+                    json.dump(capabilities, file, indent=4)
+                    file.flush()
```

Comment:
Hi Till. Basically we didn't want to call the capabilities API for each of the header_field or line_item_field, as it would take more time. That's why we decided to store the capabilities temporarily on the disk. If there is any other way of saving time, can you suggest?